### DISCUSSION TITLE
Data Analysis about peptide-labeled protein LC-MS/MS

### DISCUSSION THREAD

**Discussion by yanze039** [OpenMS/OpenMS#6976] [Q&A]
Hi,

I am a beginner to MS data analysis. I am asking for help to achieve the task below.
I have a protein (BSA), where many residues are randomly labeled by a known peptide. The linker is known that crosslinking often uses. I have the LC-MS/MS data of this molecule (Trypsin degist). Now I want to use OpenMS to analysis this data to find which residues of the protein are labeled by the peptide, i.e. to find the label positions.  (I really would like to use pythonic code or command line to achieve it.)

Does anyone has a good idea about how to do that?

Thanks in advance!!

**Comment by jpfeuffer** [comment]
Hi!

I am not a XL expert but I would start looking into OpenPepXL as a command line tool.
You might be able to emulate what you want to do by creating a fasta database containing BSA and the peptide as a second "protein".

If that does not work you might need to use the underlying algorithms that OpenPepXL applies in your own way. To check this you could look at the OpenPepXL.cpp and look at which classes are used there.
You would then need to check which of them are available in pyopenms by checking the API documentation.

**Comment by enetz** [comment]
If I understand it correctly, you have BSA cross-linked with a commonly used cross-linker, but not to itself but to another specific known peptide?
That should be possible to do with OpenPepXLLF (OpenPepXL expects isotopically labeled cross-linkers). Put BSA and that other peptide in a fasta file as two proteins. The parameters of OpenPepXLLF allow you to define the cross-linker by mass and linked residue types. The default settings describe DSS (and also work for BS3). For FDR estimation, don't forget to add reverse sequence decoys to the fasta file (e.g. with the OpenMS tool DecoyDatabase) and run XFDR on the OpenPepXLLF output.
This shouldn't require any modification of the algorithm, unless I misunderstood the data.
